### ISSUE TITLE
False positive "no-member" for Regex flags

### ISSUE THREAD

**Issue by piehld** [PyCQA/pylint#5873]
### Bug description

Pylint returns false positive for the regex flag `re.I`, failing with: 

```
JournalTitleAbbreviationProvider.py:164:149: E1101: Module 'regex' has no 'I' member; maybe 'i'? (no-member)
```

Link to example code: [Line 164 here](https://github.com/rcsb/py-rcsb_utils_citation/blob/84ab9b271ce28d7ecd9173c2919d59aa74936b68/rcsb/utils/citation/JournalTitleAbbreviationProvider.py#L164)

I suspect this issue is related to a recent update in the regex library (in version 2022.3.2, released last week). See the changes in `regex_3/_regex_core.py` in the [commit here](https://github.com/mrabarnett/mrab-regex/commit/67ab7171969a5c0562ee8f84595d78de0633ccb2#diff-ff8a8be7da8c344a21b087a89af61ae465c87a35d601e17acfca78fc7ef19e83).

If the source of the issue is on the regex end, let me know and I can post the issue on their repository there. I apologize I'm not familiar enough with either the pylint or regex codebases to readily determine what the true root of the issue is. 

### Command used

```shell
pylint --disable=R,C --reports=n JournalTitleAbbreviationProvider.py
```

### Pylint output

```shell
************* Module JournalTitleAbbreviationProvider
JournalTitleAbbreviationProvider.py:164:149: E1101: Module 'regex' has no 'I' member; maybe 'i'? (no-member)

------------------------------------------------------------------
Your code has been rated at 9.80/10 (previous run: 9.02/10, +0.78)
```

### Expected behavior

There shouldn't be an error—regex does indeed have flag member "I".

### Pylint version

```shell
pylint==2.12.2
Python 3.9.7
```

### OS / Environment

Mac OSX, Linux

### Additional dependencies
```
regex==2022.3.2

# Others unrelated to issue in question, but required for reproducing on example file:
rcsb.utils.io >= 1.15
nltk >= 3.2.5
```

**Comment by julianstirling** [comment]
It seems to be something weird that regex is now doing because I can install very old versions of pylint and this bug persists. I have fixed it by setting my version of regex to `regex==2022.1.18`. However, there does seem to be something fishy as the flags do indeed exists when you are in Python.

**Comment by DanielNoord** [comment]
This no longer reproduces on `main`. 

```console
❯ pylint --version
pylint 2.15.0-dev0
astroid 2.12.0-dev0
Python 3.10.0 (default, Oct 27 2021, 16:20:25) [Clang 13.0.0 (clang-1300.0.29.3)]
❯ pylint test.py
cat ************* Module test
test.py:3:6: E1101: Module 're' has no 'D' member; maybe 'A'? (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)                                                                                                 
❯ cat test.py
import re
print(re.I)
print(re.D)
```

**Comment by mbyrnepr2** [comment]
@DanielNoord I think there is a bit of a mixup here. The issue is because of the `regex` library not `re`.
In the linked example, regex is aliased to `re`.

Here is a quick reproducer:
```python
pip install regex
```

```python
import regex

print(regex.I)
```

example.py:3:6: E1101: Module 'regex' has no 'I' member; maybe 'i'? (no-member)

**Comment by DanielNoord** [comment]
Sorry, my bad! We should add a `regex` brain or allow the `re` brain to also catch `regex` which might actually work as well.

**Comment by smontanaro** [comment]
@DanielNoord Coming late to the game as I just stumbled on this problem. Looking at ``astroid/brain/brain_re.py`` it seems to me that ``regex`` has more flags than ``re``. I [opened an issue](https://github.com/PyCQA/astroid/issues/1911) over there with a workable change.

**Comment by mbyrnepr2** [comment]
Closing as this is now fixed by https://github.com/PyCQA/astroid/pull/1934.
